backend/upvote/router.ts

```diff
@@ -0,0 +1,109 @@
+import type { Request, Response, NextFunction } from "express";
+import { Types } from "mongoose";
+import express from "express";
+
+import PlaceRequestCollection from "../request/collection";
+import { constructPlaceRequestResponse } from "../request/util";
+import UserCollection from "../user/collection";
+import * as userMiddleware from "../user/middleware"
+const router = express.Router();
+
+/**
+ * @name POST /api/upvotes/{requestId}
+ */
+router.post(
+  "/:requestId",
+
+  async (req: Request, res: Response) => {
+    const request = await PlaceRequestCollection.findOneById(
+      req.params["requestId"] as string
+    );
+    if (!request) {
+      res.status(403).json({
+        message: `Request with id ${req.params["requestId"]} does not exist`,
+      });
+      return;
+    }
+    if (!req.session.userId) {
+      res.status(403).json({
+        message: "User is not logged in.",
+      });
+      return;
+    }
+
+    const sessionUser = await UserCollection.findOneFromGapiUserId(req.session.userId);
+    if (sessionUser === null){
+      res.status(500).json({ //should never reach here
+        message: "Session user cannot be found"
+      });
+      return;
+    };
+    if (request.upvotingUsers.includes(sessionUser?._id)) {
+      // this should not happen
+      res.status(201).json({
+        message: "User has already upvoted this response",
+        request: request,
+      });
+    } else {
+      request.upvotingUsers.push(sessionUser?._id);
```

Comment:
after this push, we need a way to repopulate `request.upvotingUsers`

Possibly we can move the check to see if the user already upvoted the request to `/request/middleware.ts` and add a function in `/request/collection.ts` like ```updateOne(requestId: string, upvotingUser: string): Promise<HydratedDocument<Request>>``` to add an "upvoter" then return the populated request object.